### PR TITLE
Fix dependency issues

### PR DESCRIPTION
**Pull request by EngineerDanny**
The [animint](https://github.com/tdhock/animint) package in the root **DESCRIPTION** file has been discontinued.
The Github CI test showed that it is not possible to install it because it could not recognize the package.
Fast forward, there is an updated version called the [animint2](https://github.com/tdhock/animint2) which comes with improved features and better syntax.

### CODE REVIEW

**Comment by EngineerDanny** [comment]
This PR changes the DESCRIPTION file by updating the **animint2** package.
I have updated, checked and tested all the functions that depend on the **animint2** package.

**Comment by EngineerDanny** [comment]
Github CI Test Before the PR : 

Screenshot 2022-06-21 at 2 39 41 PM

Github CI Test After initial PR :

Screenshot 2022-06-21 at 2 42 24 PM

**git2r** package is the only package left to be updated.

**Comment by EngineerDanny** [comment]
[git2r](https://github.com/ropensci/git2r) update PR #3 has been merged into this branch since they all address dependency conflicts.

**Comment by EngineerDanny** [comment]
This PR also focuses on updating the [git2r](https://github.com/ropensci/git2r) package to support the latest features and improvements.
More specifically, `git2r (>= 0.30.1)`
However, updating the package came with a minor hurdle.
The package has transitioned from supporting S4 objects to S3 objects.
Hence, it has made some of the functions used with `git2r (<= 0.21.0)` obsolete.
Therefore, I have updated and tested all the git2r functions that we use to support S3 objects.

This PR fixes issues [#44 ](https://github.com/analyticalmonk/Rperform/issues/44) and [#45 ](https://github.com/analyticalmonk/Rperform/issues/45)

**Comment by EngineerDanny** [comment]
Explicitly writing scripts to install dependencies is not widely accepted in the R community since it makes the **DESCRIPTION** file redundant.
Therefore, [r-lib/actions/setup-r-dependencies](https://github.com/r-lib/actions/tree/v2/setup-r-dependencies) is adopted to automatically install packages declared in **DESCRIPTION**.

The Github CI test is now successful ✅.

**Comment by tdhock** [comment]
binary files like .github/actions/.DS_Store should not be added to the github repo. please remove them from the PR and add .DS_Store to the .gitignore file.

File: .github/actions/test-info/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Install Rperform with its dependendcies
+Rscript -e "install.packages(c('ggplot2', 'devtools', 'remotes', 'testthat', 'microbenchmark'), repos='http://cran.us.r-project.org')"
```
Comment (tdhock) [inline .github/actions/test-info/entrypoint.sh:4]: please delete this and use r-lib/actions/install-pkg-dependencies instead

File: R/plot_metrics.R [hunk 2]
```diff
@@ -75,12 +75,12 @@ plot_metrics <- function(test_path, metric, num_commits = 5, save_data = FALSE,
...
   stopifnot(length(save_plots) == 1)
   floor(num_commits)
-  
+
   if (metric == "time") {
     if (interactive) {
       temp_out <- capture.output(.plot_interactive_time(test_path, num_commits, save_data, save_plots))
     } else {
-      temp_out <- capture.output(.plot_time(test_path, num_commits, save_data, save_plots))
+       temp_out <- capture.output(.plot_time(test_path, num_commits, save_data, save_plots))
```
Comment (tdhock) [inline R/plot_metrics.R:83]: please make indentation consistent (here there are three spaces for else but only two for if)

**Review by tdhock — CHANGES_REQUESTED** [review]

File: .github/actions/test-info/entrypoint.sh (hunk 1 above)
Comment (EngineerDanny) [reply .github/actions/test-info/entrypoint.sh:4]: I have actually refactored and deleted this file and used **r-lib/actions/install-pkg-dependencies** on the **test-pkg** branch on the main [PR](https://github.com/EngineerDanny/Rperform/pull/1). I will close this PR.

File: R/plot_metrics.R (hunk 2 above)
Comment (EngineerDanny) [reply R/plot_metrics.R:83]: Sure. I will resolve the indentation and push on the **test-pkg** branch.

Comment (EngineerDanny) [reply R/plot_metrics.R:83]: This issue has been resolved on the **test-pkg** branch.
I used R studio's indentation to address all the indentation issues in all the R files.
You can review the main [PR](https://github.com/EngineerDanny/Rperform/pull/1)

**Pull request closed without merging**